extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -298,22 +270,23 @@ public void triggerEvent(String channelId, String event) {
     }
 
     /**
-     * Returns the scheduler for the Astro jobs
-     */
-    public ExpressionThreadPoolExecutor getScheduler() {
-        return scheduledExecutor;
-    }
-
-    /**
      * Adds the provided {@link Job} to the queue (cannot be {@code null})
      *
      * @return {@code true} if the {@code job} is added to the queue, otherwise {@code false}
      */
-    public boolean addJobToQueue(Job job) {
-        if (job != null) {
-            return scheduledJobs.add(job);
+    public void schedule(Job job, Calendar eventAt) {
+        monitor.lock();
+        try {
+            long sleepTime = eventAt.getTimeInMillis() - new Date().getTime();
```

Comment:
I see two approaches: 

* we maybe can handle this easily as the astro calculations internally use julian dates, which is not subject to such anomalies - so it would be easiest to get the amount of milliseconds to wait directly from there instead of messing around with Date/Calendar objects which for this purpose actually are not really helpful and just create chaos. So it could be that we were better off staying within JD and not taking this detour. But I have to admit that I did not yet focus in this.

* Once the calculations get fixed in the scheduler, I would be happy to re-use the same utility method to calculate the time difference here. But I still do not see that this would also require using the _expression_ based scheduling as a whole. Maybe it would make sense to create such utility methods in the ScheduledExecutorService that our ThreadPoolManager returns.

So while I agree that we need to fix it and only do it once, I don't see that it conflicts with this.